mapentity/views/generic.py

```diff
@@ -356,6 +356,38 @@ def form_invalid(self, form):
         return super().form_invalid(form)
 
 
+class MapEntityDuplicate(ModelViewMixin, SingleObjectMixin, View):
+    http_method_names = ['post', ]
+
+    @classmethod
+    def get_entity_kind(cls):
+        return mapentity_models.ENTITY_DUPLICATE
+
+    @classmethod
+    def get_title(cls):
+        return cls.model.get_create_label()
+
+    @view_permission_required()
+    def dispatch(self, *args, **kwargs):
+        return super().dispatch(*args, **kwargs)
+
+    @view_permission_required()
+    def post(self, request, *args, **kwargs):
+        original_object = self.get_object()
+        try:
+            clone = original_object.duplicate(request=request)
+            if not clone:
+                messages.error(self.request, _("Duplication is not available for this object"))
+                return HttpResponseRedirect(original_object.get_detail_url())
+            log_action(self.request, clone, ADDITION)
+            messages.success(self.request,
+                             _(f"{self.get_object()._meta.verbose_name} has been duplicated successfully"))
```

Comment:
don't put formated values in translated text
```suggestion
                             f"{self.get_object()._meta.verbose_name} " + _(f"has been duplicated successfully"))
```
or something like that